src/commands/init.ts

```diff
@@ -98,24 +80,48 @@ export abstract class InitCommand extends BaseCommand {
       this.error(chalk.red(`${err}\nYour docker compose file at ${from_path} was unable to be converted to an Architect component. If you think this is a bug, please submit an issue at https://github.com/architect-team/architect-cli/issues.`));
     }
 
-    fs.writeFileSync(flags['component-file'], architect_yml);
-    this.log(chalk.green(`Converted ${path.basename(from_path)} and wrote Architect component config to ${flags['component-file']}`));
+    fs.writeFileSync(component_file, architect_yml);
+    this.log(chalk.green(`Converted ${path.basename(from_path)} and wrote Architect component config to ${component_file}`));
     this.log(chalk.blue('The component config may be incomplete and should be checked for consistency with the context of your application. Helpful reference docs can be found at https://docs.architect.io/components/architect-yml.'));
   }
 
-  async getComposeFromPath(flags: any): Promise<string> {
-    let from_path;
+  async run(): Promise<void> {
+    const { flags, args } = await this.parse(InitCommand);
+
+    const compose_exist = await this.doesDockerComposeYmlExist();
+    if (!args.name) {
+      const answer: { chosen: any } = await inquirer.prompt([
+        {
+          name: 'chosen',
+          message: `Please provide a name for your ${compose_exist ? 'component' : 'project'}:`,
+          type: 'input',
+        }]);
+      args.name = answer.chosen;
+    }
+
+    if (!ComponentSlugUtils.Validator.test(args.name)) {
+      if (flags['from-compose']) {
+        throw new Error(`Component name can only contain lowercase letters and dashes, and must start and end with a letter.`);
+      }
+      throw new Error(`Project name can only contain lowercase letters and dashes, and must start and end with a letter.`);
+    }
```

Comment:
The UX is pretty jarring here - if the user doesn't create a project that adheres to the standards, it errors out and they have to run `init` again. Two things to add to this:

1. Don't Error - just warn if they don't follow the rules and allow them to re-name their project
2. Give a hint before prompting. Something like:
`Please provide a name for your project. \n use only lowercase letters and dashes, and do not end or start with a dash`

Reply:
Thanks for the review, Zach! I will work on that.